File: flutter-application-main/lib/models/rally/stage.dart
import 'package:dart_mappable/dart_mappable.dart';
import 'package:gibz_mobileapp/models/rally/rally_models.dart';

@MappableClass()
class Stage {
  final String id;
  final String title;
  final StageInformation preArrivalInformation;
  final StageInformation information;
  final List<Location> locations;
  final StageActivity stageActivity;

  Stage({
    required this.id,
    required this.title,
    required this.preArrivalInformation,
    required this.information,
    required this.locations,
    required this.stageActivity,
  });
}

File: frontend-main/src/models/rally/stage.ts
import { Location } from "./location";
import { RallyAttachment } from "./rallyAttachment";

export class Stage {
  id: string;
  title: string;
  preArrivalInformation: StageInformation;
  information: StageInformation;
  locations: Location[];

  constructor(
    id: string,
    title: string,
    preArrivalInformation: StageInformation,
    information: StageInformation,
    locations: Location[]
  ) {
    this.id = id;
    this.title = title;
    this.preArrivalInformation = preArrivalInformation;
    this.information = information;
    this.locations = locations;
  }
}

export class StageInformation {
  id: string;
  title: string;
  content: string;
  attachments: RallyAttachment[];

  constructor(
    id: string,
    title: string,
    content: string,
    attachments: RallyAttachment[]
  ) {
    this.id = id;
    this.title = title;
    this.content = content;
    this.attachments = attachments;
  }
}
